subsys/logging/log_output_syst.c

```diff
@@ -786,27 +786,97 @@ void log_output_msg2_syst_process(const struct log_output *output,
 	uint8_t *data = log_msg2_get_package(msg, &len);
 
 	if (len) {
-		char *buf = data, *fmt;
+		/*
+		 * The package format:
+		 *
+		 * [0]: Argument list size including package header
+		 *      (sizeof(uint8_t *)) and format string pointer,
+		 *      in number of uint32_t words.
+		 * [1]: Number of appened strings in package.
+		 *
+		 * [sizeof(uint8_t *)]: Pointer to format string.
+		 *                      Can be NULL so needs further
+		 *                      processing.
+		 *
+		 * [sizeof(uint8_t *) * 2]: Start of arguments.
+		 *
+		 * [[0] * sizeof(uint32_t)]: Start of appended strings.
+		 */
+
+		uint8_t num_strings = data[1];
```

Comment:
This algorithm is already present in cbprintf https://github.com/zephyrproject-rtos/zephyr/blob/fdc25cd44c2b693aecff96d3dfcd0ee088d48d5b/lib/os/cbprintf_packaged.c#L602
but it is a part of formatting function. Seems like it would be worth to extract it and make public. Not sure what would be a good name for it. Something like `cbprintf_locate` or `cbrprintf_settle`?

It could be called by logging core before passing message to the backends.